Directory.Build.props

```diff
@@ -20,6 +20,7 @@
     <AdditionalFiles Include="$(MSBuildThisFileDirectory)CodeAnalysis\BannedSymbols.txt" />
   </ItemGroup>
   <PropertyGroup Label="Code Analysis">
+    <EnforceCodeStyleInBuild Condition="'$(EnforceCodeStyleInBuild)' == ''">true</EnforceCodeStyleInBuild>
```

Comment:
I am not sure we can warrant this being always on.

```md

## master

Clean: 14.32s 13.90s
No change: 2.93s 3.01s
Change: 10.66s 10.79s

Desktop no change: 1.64s 1.66s
Desktop change: 7.2s 7.47s

## this pr

Clean: 19.43s 17.32s
No change: 2.97s 3.07s
Change: 13.98s 14.13s

Desktop no change: 1.62s 1.63s
Desktop change: 9.06s 9.18s
```

For full project builds, that's 30% overhead. For desktop builds, 24%. I don't want this kind of overhead on my builds. So if we're implementing this it needs to either be off by default, or there needs to be an easy way to turn it off.

Reply:
For the record I'm fine with only enabling this on CI. I'd like to think it shouldn't be actually needed very much on desktop, as any decent IDE (*side-eyes vs code*) should be able to display the inspections in real time as errors.

Reply:
VS will skip analyzers when hitting F5 to debug, but other IDEs can be different.
I didn't get a measurable difference at framework repo, but if there is as you shown, we can just enable it in CI, and probably put under code quality leg.